### PR TITLE
Fix specs and deprecations

### PR DESCRIPTION
**Pull request by thijsc**
Some spec failures and deprecations warnings snuck in the last months, fixing them.

### CODE REVIEW

**Comment by thijsc** [comment]
cc @mensfeld: I'm trying to get the specs green and reliable again so we can get pull requests moving quickly again.

**Comment by mensfeld** [comment]
@thijsc thanks. I plan to get back to my work on the per-instance instrumentation next week.

I appreciate your work a lot :pray:

**Comment by thijsc** [comment]
Green, merging

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -8,27 +8,58 @@
...
-def rdkafka_config(config_overrides={})
-  config = {
+def rdkafka_base_config
+  {
     :"api.version.request" => false,
     :"broker.version.fallback" => "1.0",
     :"bootstrap.servers" => "localhost:9092",
-    :"group.id" => "ruby-test-#{Random.new.rand(0..1_000_000)}",
-    :"auto.offset.reset" => "earliest",
-    :"enable.partition.eof" => false
```
Comment (geoff2k) [inline spec/spec_helper.rb:19]: It seems like there are two main changes in this PR-- the 60 second timeout in the `around(:each)`  on line 130, and the extraction of this configuration out of the default, therefore moving it out of the producer configuration.

Were these settings interfering with some of the producer-related tests?

**Pull request merged**